projects/core/src/site-context/services/currency-initializer.ts

```diff
@@ -0,0 +1,58 @@
+import { Injectable } from '@angular/core';
+import { concat, defer, Observable, Subscription } from 'rxjs';
+import { tap } from 'rxjs/operators';
+import { ConfigInitializerService } from '../../config';
+import { getContextParameterDefault } from '../config/context-config-utils';
+import { SiteContextConfig } from '../config/site-context-config';
+import { CurrencyService } from '../facade';
+import { CURRENCY_CONTEXT_ID } from '../providers';
+import { CurrencyStatePersistenceService } from './currency-state-persistence.service';
+
+@Injectable({ providedIn: 'root' })
+export class CurrencyInitializer {
+  protected subscription: Subscription;
+
+  constructor(
+    protected currencyService: CurrencyService,
+    protected currencyStatePersistenceService: CurrencyStatePersistenceService,
+    protected configInit: ConfigInitializerService
+  ) {}
+
+  /**
+   * Initializes the value of the active currency.
+   */
+  initialize(): void {
+    this.subscription = concat(
```

Comment:
Can we add ngOnDestroy to dispose the subscription?

It the default impl. all  the inner observables complete right after emission so no memory leaks are expected. But ngOnDestroy is needed just in case a customer customizes one of those observables and doesn't remember to complete it. It's not obvious, that's why we put it just in case.

Reply:
It makes sense. I actually forgot to add it.